cirq-google/cirq_google/engine/engine_validator.py

```diff
@@ -0,0 +1,172 @@
+          sweeps and runs, this will be a single circuit.  For batches,
+          this will be a list of circuits.
+        sweeps:  Parameters to run with each circuit.  The length of the
+          sweeps sequence should be the same as the circuits argument.
+        repetitions:  Number of repetitions to run with each sweep.
+        gate_set:  Serializer to use to serialize the circuits and sweeps.
+        max_size:  proto size limit to check against.
+
+    Raises:
+        RuntimeError: if compiled proto is above the maximum size.
+    """
+    batch = v2.batch_pb2.BatchProgram()
+    packed = any_pb2.Any()
+    for circuit in circuits:
+        gate_set.serialize(circuit, msg=batch.programs.add())
+    packed.Pack(batch)
+    message_size = len(packed.SerializeToString())
+    if message_size > max_size:
+        raise RuntimeError("INVALID_PROGRAM: Program too long.")
+
+
+def create_gate_set_validator(max_size: int = MAX_MESSAGE_SIZE) -> GATE_SET_VALIDATOR_TYPE:
+    """Creates a Callable gate set validator with a set message size.
+
+    Args:
+        max_size:  proto size limit to check against.
+
+    Returns: Callable to use in validation with the max_size already set.
+    """
+
+    def _validator(
+        circuits: Sequence[cirq.AbstractCircuit],
+        sweeps: Sequence[cirq.Sweepable],
+        repetitions: int,
+        gate_set: Serializer,
+    ):
+        return validate_gate_set(circuits, sweeps, repetitions, gate_set)
+
+    return _validator
+
+
+def validate_for_engine(
+    circuits: Sequence[cirq.AbstractCircuit],
+    sweeps: Sequence[cirq.Sweepable],
+    repetitions: Union[int, List[int]],
+    max_moments: int = MAX_MOMENTS,
+    max_repetitions: int = MAX_TOTAL_REPETITIONS,
+    max_duration_ns: int = 55000,
+) -> None:
+    """Validate a circuit and sweeps for sending to the Quantum Engine API.
+
+    Args:
+       circuits:  A sequence of  `cirq.Circuit` objects to validate.  For
+          sweeps and runs, this will be a single circuit.  For batches,
+          this will be a list of circuits.
+       sweeps:  Parameters to run with each circuit.  The length of the
+          sweeps sequence should be the same as the circuits argument.
+       repetitions:  Number of repetitions to run with each sweep.
+       max_moments: Maximum number of moments to allow.
+       max_repetitions: Maximum number of parameter sweep values allowed
+           when summed across all sweeps and all batches.
+       max_duration_ns:  Maximum duration of the circuit, in nanoseconds.
+    """
+    _verify_reps(sweeps, repetitions, max_repetitions)
+    _validate_depth(circuits, max_moments)
+    _verify_measurements(circuits)
+
+
+def create_engine_validator(
+    max_moments: int = MAX_MOMENTS,
+    max_repetitions: int = MAX_TOTAL_REPETITIONS,
+    max_duration_ns: int = 55000,
+) -> VALIDATOR_TYPE:
+    """Creates a Callanle gate set validator with a set message size.
```

Comment:
callanle -> callable

Reply:
this is a matter of style, but I would find it enormously helpful if these factory functions gave some hint at where you could pass the result to:

"""Creates a validator...

This can be a useful helper for generating validators for `cg.ValidatingSampler`'s `validator` argument
"""

Reply:
Good idea.  Done. Let me know if the additional paragraph is still not clear.